application/src/ext/java/org/opentripplanner/ext/fares/model/FareTransferRule.java

```diff
@@ -7,27 +7,104 @@
 import javax.annotation.Nullable;
 import org.opentripplanner.model.fare.FareProduct;
 import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.utils.tostring.ToStringBuilder;
 
-public record FareTransferRule(
-  FeedScopedId id,
-  @Nullable FeedScopedId fromLegGroup,
-  @Nullable FeedScopedId toLegGroup,
-  int transferCount,
-  @Nullable Duration timeLimit,
-  Collection<FareProduct> fareProducts
-) {
-  public FareTransferRule {
-    Objects.requireNonNull(id);
-    fareProducts = List.copyOf(fareProducts);
-  }
-  public String feedId() {
-    return id.getFeedId();
+public final class FareTransferRule {
```

Comment:
The conversion of `FareTransferRule` to a class might have broken serialization. I got this error after building the graph and reading it:
```
14:28:12.884 WARN [main]  (SerializedGraphObject.java:214) Deserialization exception while loading graph: /~/data/trimet/graph.obj
org.objenesis.ObjenesisException: java.io.NotSerializableException: class org.opentripplanner.ext.fares.model.FareTransferRule not serializable
Serialization trace:
fareTransferRules (org.opentripplanner.ext.fares.impl.gtfs.CombineInterlinedLegsFactory)
fareServiceFactory (org.opentripplanner.routing.graph.SerializedGraphObject)
```